addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -133,6 +133,8 @@ def portal_my_opportunities(self, page=1, date_begin=None, date_end=None, sortby
         if date_begin and date_end:
             domain += [('create_date', '>', date_begin), ('create_date', '<=', date_end)]
         # pager
+        leads_sudo = CrmLead.sudo().search(domain)
+        domain = [('id', 'in', leads_sudo.ids)]
```

Comment:
as far as I can remember, this should work too (and do only one query)

```suggestion
        leads_sudo = CrmLead.sudo()._search(domain)
        domain = [('id', 'in', leads_sudo)]
```

Reply:
Indeed it should work

Reply:
Thought about it but effectively it means you're doing the same query twice. Once for the count, and once again for the search. And since it's a subquery I'm not sure the limit will apply so well.

Shouldn't matter too much though